lib/components/user/notification-prefs-pane.js

```diff
@@ -85,28 +146,236 @@ const NotificationPrefsPane = ({
+      phoneFormatOptions,
+      touched // Formik prop
+    } = this.props
+    const { isEditing, newPhoneNumber } = this.state
+    const isPending = !isBlank(initialPhoneNumber) && !initialPhoneNumberVerified
+
+    // Here are the states we are dealing with:
+    // - First time entering a phone number/validation code (blank value, not modified)
+    //   => no color, no feedback indication.
+    // - Typing backspace all the way to erase a number/code (blank value, modified)
+    //   => red error.
+    // - Typing a phone number that doesn't match the configured phoneNumberRegEx
+    //   => red error.
+    const isPhoneInvalid = !isPossiblePhoneNumber(newPhoneNumber)
+    const showPhoneError = isPhoneInvalid && !isBlank(newPhoneNumber)
+    const phoneErrorState = showPhoneError ? 'error' : null
+    const codeErrorState = getErrorState(this.props, 'validationCode')
+
+    return (
+      <>
+        {isEditing
+          ? (
+            // FIXME: If removing the Save/Cancel buttons on the account screen,
+            // make this <FormGroup> a <form> and remove onKeyDown handler.
+            <FormGroup validationState={phoneErrorState}>
+              <ControlLabel>Enter your phone number for SMS notifications:</ControlLabel>
+              <ControlStrip>
+                <InlinePhoneInput
+                  className='form-control'
+                  country={phoneFormatOptions.countryCode}
+                  onChange={this._handleNewPhoneNumberChange}
+                  onKeyDown={this._handlePhoneNumberKeyDown}
+                  placeholder='Enter phone number'
+                  type='tel'
+                  value={newPhoneNumber}
+                />
+                <Button
+                  bsStyle='primary'
+                  disabled={isPhoneInvalid}
+                  onClick={this._handleRequestCode}
+                >
+                  Send verification text
+                </Button>
+                { // Show cancel button only if a phone number is already recorded.
+                  initialPhoneNumber && <Button onClick={this._handleCancelEditNumber}>Cancel</Button>}
+                {showPhoneError && <HelpBlock>{INVALID_PHONE_MSG}</HelpBlock>}
+              </ControlStrip>
+            </FormGroup>
+          ) : (
+            <FormGroup>
+              <ControlLabel>SMS notifications will be sent to:</ControlLabel>
+              <ControlStrip>
+                <InlineStatic>
+                  {formatPhoneNumber(initialPhoneNumber)}
+                  {' '}
+                  {isPending
+                    // eslint-disable-next-line jsx-a11y/label-has-for
+                    ? <Label bsStyle='warning'>Pending</Label>
+                    // eslint-disable-next-line jsx-a11y/label-has-for
+                    : <Label bsStyle='success'>Verified</Label>
+                  }
+                </InlineStatic>
+                <Button onClick={this._handleEditNumber}>Change number</Button>
```

Comment:
There is no way to delete a phone number. Not sure if needed since a user could switch to either email or do not notify.